src/Numerical/Spectral/MeanValue.hpp

```diff
@@ -0,0 +1,41 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function MeanValue.
+
+#pragma once
+
+#include "Domain/Side.hpp"
+#include "Numerical/Spectral/DefiniteIntegral.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+/*! \ingroup Functors
+ * \brief Compute the mean value of a grid-function over a manifold.
+ *
+ * \remarks The mean value is computed on the reference element(s).
+ *
+ * \returns the mean value of f on the manifold
+ */
+template <size_t Dim>
+double mean_value(const Scalar<DataVector>& f, const Index<Dim>& extents) {
+  return Basis::lgl::definite_integral(f, extents) / two_to_the(Dim);
+}
+
+/*! \ingroup Functors
+ *
+ * Compute the mean value of a grid-function on a boundary of a manifold.
+ * \f$mean value = \int f dV / \int dV\f$
+ *
+ * \remarks The mean value is computed on the reference element(s).
+ *
+ * \returns the mean value of f on the boundary of the manifold
+ *
+ * \param f the grid function of which to find the mean.
+ * \param extents the extents of the manifold on which f is located.
+ * \param d the dimension which is sliced away to get the boundary.
+ * \param side whether it is the lower or upper boundary in the d-th dimension.
+ */
+template <size_t Dim>
+double mean_value_on_boundary(const DataVector& f, const Index<Dim>& extents,
```

Comment:
Why does this take a DataVector and the `mean_value` function takes a `Scalar<DataVector>`?

Reply:
I'd suggest these functions (and `definite_integral`) take a simple `DataVector`... e.g. one might want to know the integral or mean of v_x as a diagnostic, which is easier if the interfaces take `DataVector`.

Reply:
:+1:

Reply:
ping

Reply:
:bell: